packages/common/src/index.ts

```diff
@@ -154,6 +154,28 @@ export const validateEmail = (email: string): boolean => {
     return re.test(String(email).toLowerCase());
 };
 
+export type PasswordValidationResult = {
+    isLengthValid: boolean;
+    hasLetter: boolean;
+    hasNumberOrSymbol: boolean;
+    isPasswordValid: boolean;
+};
+
+export function validatePassword(password: string): PasswordValidationResult {
+    const minLength = 8;
+    const hasLetter = /[a-zA-Z]/.test(password);
+    const hasNumberOrSymbol = /[\d\W_]/.test(password);
+    const isLengthValid = password.length >= minLength;
+    const isPasswordValid = isLengthValid && hasLetter && hasNumberOrSymbol;
+
+    return {
+        isLengthValid,
+        hasLetter,
+        hasNumberOrSymbol,
+        isPasswordValid,
+    };
+}
```

Comment:
if we rewrite this using ZOD we can simplify a lot of frontend code. mantine integrates with zod nicely and we already use that here for example `packages/frontend/src/components/LimitButton/LimitForm.tsx`

it will simplify this chunk of code as well:

```
const passwordSchema = z.object({
  password: z
    .string()
    .min(8) 
    .regex(/[a-zA-Z]/, { message: 'must contain a letter' })
    .regex(/[\\d\\W_]/, { message: 'must contain a number or symbol' })
});
```

Reply:
Will look into that.

Reply:
Hey @IrakliJani , It's good only but it shows error one by one right ? first it check for size and until size is okay it's shows same error and then it shows next error and then next error, it's like three warnings one by one. If the user enters first eight chars and then he sees that he need to add one number then he thinks and then he adds one number or symbol then there is other error which says atleast one char is needed. Won't it be like halting the user three times and increasing the number of clicks as those form validation get's checked after submission. Instead we can use `PasswordInput` but that won't decrease the code lines instead add some more. 

https://mantine.dev/core/password-input/#strength-meter-example

But with that previous code we can just implement show the PasswordMessages only when input value length. So two options:

1. Adding condition with the previous code.
2. Keeping this schema validation and then also that strength so that user while typing can see the constraints and can choose password accordingly

So which one are we going with? I am done with the schema validation part along with unit test's just, can go with anything as per the requirement.

https://github.com/lightdash/lightdash/assets/73348574/e5992fab-0f6a-4f96-b468-c8a7f8e818e9

Reply:
I see... that banner with password strength looks pretty nice.
we should probably ignore UI related changes in the PR and go with what's been implemented so far

tagging @PriPatel since this is part of user's onboarding experience

Reply:
@iampranavdhar You are right - we should not show the user errors 1 by 1. 
If there are 3 errors, all errors must be shown at the same time. 

Thank you so much!

Reply:
@iampranavdhar, what you already have is great. let's skip banner as well, we'll create a separate ticket for it and do that as a separate PR. thanks @iampranavdhar 🙌

Reply:
Hello @IrakliJani  and @PriPatel Tried that strength along with schema validations. This is how it looks. I can push if this looks good.

Uploading Screen Recording 2023-08-08 at 11.51.49 PM.mov…

Reply:
looks great @iampranavdhar 🙌

Reply:
Pushed 🚀 @IrakliJani

Reply:
Yeah @iampranavdhar looks awesome. Thanks so much for this!

Reply:
testing this locally now

Reply:
hey @iampranavdhar banner items and progress bar is not in sync, can you have a look?

https://github.com/lightdash/lightdash/assets/962095/5266cbfd-887e-4332-aa1c-061a955b80a3

also, let me know if you need help, happy to take this PR from here

Reply:
@IrakliJani and @owlas Can you check it now?

https://github.com/lightdash/lightdash/assets/73348574/52e5a529-27f4-400f-97cd-e5d2c8dbcf44

Reply:
found few issues with the view. Im gonna push a fix now